tests/ui/codegen/virtual-function-elimination.rs

```diff
@@ -0,0 +1,17 @@
+//@ build-pass
+//@ compile-flags: -Zvirtual-function-elimination=true -Clto=true
+//@ only-x86_64
+//@ no-prefer-dynamic
```

Comment:
why is this argument needed?

Reply:
`./x test` needs this argument to perform LTO.
Below is the error message without `no-prefer-dynamic`:
```
running 1 tests

[ui] tests/ui/codegen/virtual-function-elimination.rs ... F


failures:

---- [ui] tests/ui/codegen/virtual-function-elimination.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/home/ray/Desktop/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/home/ray/Desktop/rust/tests/ui/codegen/virtual-function-elimination.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/home/ray/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/home/ray/Desktop/rust/vendor" "--sysroot" "/home/ray/Desktop/rust/build/x86_64-unknown-linux-gnu/stage1" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "/home/ray/Desktop/rust/build/x86_64-unknown-linux-gnu/test/ui/codegen/virtual-function-elimination" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/home/ray/Desktop/rust/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/home/ray/Desktop/rust/build/x86_64-unknown-linux-gnu/test/ui/codegen/virtual-function-elimination/auxiliary" "-Zvirtual-function-elimination=true" "-Clto=true"
stdout: none
--- stderr -------------------------------
error: cannot prefer dynamic linking when performing LTO
   |
   = note: only 'staticlib', 'bin', and 'cdylib' outputs are supported with LTO

error: aborting due to 1 previous error
------------------------------------------
```